File: js/tests/maxMinNumberArray.test.js
const diffMaxMin = require("../maxMinNumberArray");

const data = [
  [[10, 4, 1, 2, 8, 91], 90],
  [[-70, 43, 34, 54, 22], 124]
];

describe("diffMaxMin", () => {
  describe("Return difference of biggest and smallest numbers from an array", () => {
    test.each(data)("diffMaxMin(%p) = %i", (arr, expected) => {
      var actual = diffMaxMin(arr);

      expect(actual).toEqual(expected);
    });
  });
});

File: CSharp/Tests/MaxMinNumberArrayTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class MaxMinNumberArrayTest
    {
        [Theory]
        [InlineData(new int[] { 10, 4, 1, 2, 8, 91 }, 90)]
        [InlineData(new int[] { -70, 43, 34, 54, 22 }, 124)]
        public void DiffMaxMin_IntArray_ReturnDiffOfMaxAndMinValuesFromArray(int[] arr, int expected)
        {
            var actual = MaxMinNumberArray.DiffMaxMin(arr);

            Assert.Equal(expected, actual);
        }
    }
}